### PR TITLE
GAME TREE SCENARIOS

### PR DESCRIPTION
**Pull request by sriharikapu**
Covers various scenarios of game tress
Implemented the following codes 

Actions included

- [x] buyer-commit
- [x] buyer-complain
- [x] buyer-redeem
- [x] buyer-refund
- [x] seller-cancel
- [x] seller-commit
- [x] seller-admitFault

Helpers included 

- [x] Delay
- [x] Formatter
- [x]  Check Balances

Also included

- [x] sign Transactions

Tests

- [x] Modularized

Scenarios Covered

- [x] SELLER LISTS A VOUCHER FOR SALE
- [x] BUYER PURCHASES A VOUCHER
- [x] BUYER REDEEMS A COMMITTED VOUCHER
- [x] BUYER CLAIMS REFUND FOR A COMMITTED VOUCHER
- [x] BUYER COMPLAINS ABOUT REDEEMED VOUCHER
- [x] BUYER COMPLAINS ABOUT REFUND VOUCHER
- [x] SELLER ACCEPTS REDEEM FAULT ON COMPLAIN
- [x] SELLER ACCEPTS REFUND FAULT ON COMPLAIN

[Read More](https://github.com/bosonprotocol/contracts/blob/gametree-trace/test/readme.md)

### CODE REVIEW

**Review by hswopeams — CHANGES_REQUESTED** [review]
A couple of things

- It looks like the formatter didn't like something. That's what made the pipeline fail. Try running `./go tests:format_fix` and then recommitting
- Please move the actions folder and everything in it into a folder called testGameTree or similar. Right now it's not clear that the actions folder is for tests
- The tests output using console.log, but I don't see any assertions. It's assertions that make a test a test. For instance when balances are checked in "Check Balances" blocks, there should be an assertion instead of a `console.log`

File: testGameTree/5-buyer-commit-refund.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const delay = require("./helpers/delay");
+const {describe,it} = require("mocha");
+
+let format = require("./helpers/formatter");
+
+describe("TEST SCENARIO 05 :: SELLER CREATES, BUYER COMMITS & BUYER REFUNDS", async function() {
+
+    let voucherSetDetails;
+    let committedVoucher;
+    let refundedVoucher;
```
Comment (HristiyanG) [inline testGameTree/5-buyer-commit-refund.js:14]: redundant variable

**Review by HristiyanG — CHANGES_REQUESTED** [review]

**Comment by HristiyanG** [comment]
Just finished my review, left some comments there, but just still some general questions: 
- Why the old test were deleted? There were roughly about 400 test. Do these scenarios cover each and every missing test?
	- if that is not the case, please bring back the old test
	- please consider writing a script, responsible for the user interactions, and leave the `test` script for the unit testing
- Agreed with Heather that the `actions` folder needs to be moved into the tests, maybe renamed to testGameTreeActions or similar
- Agreed with Heather that the tests should be using assertions
- Do we need to test those actions over a testnet at all?
	- we can do it by running 2 different scripts then, 1 for testnet (e.g. Rinkeby), 1 locally
	- this way while run locally, we should not bother to fill in the config file. We can get the accounts from the local rpc, deploy a contract and fill in the required params automatically
- There should be something at the end indicating that the tests are completed. Currently when the process finishes I was misled that I am doing something wrong as there is not anything which prompts me how tests went
- Maybe a little more detailed README file would be good. I don't think the delay should be that long (30000) if we test the flow locally. Maybe even we could move it into the configs?
- `constant.js` exported from the `test` folder, and now moved to actions/helpers/constants. Do we need to do that. IMHO this should be in the test folder as it's own purpose is to serve the tests
- left some comments for removing the empty lines, commented lines and other

File: testGameTree/1-seller-create.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const sellerCreate = require("./seller/createVoucher");
+const checkBalance = require("./helpers/checkBalance");
+const {describe,it} = require("mocha");
+let format = require("./helpers/formatter")
+
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/1-seller-create.js]: removed it

File: testGameTree/buyer/refundVoucher.js [hunk 3]
```diff
@@ -47,29 +47,32 @@ function refundVoucher(_voucherID) {
...
 }
 
-(async function newOrder () {
-    await refundVoucher("57896044618658097711785492504343954004559654357715190152841577105831485243398");
-})();
+// (async function newOrder () {
+//     await refundVoucher("57896044618658097711785492504343954004559654357715190152841577105831485243398");
+// })();
 
-// module.exports = RefundVoucherETHETH;
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/refundVoucher.js]: removed them

File: testGameTree/buyer/refundVoucher.js [hunk 4]
```diff
@@ -47,29 +47,32 @@ function refundVoucher(_voucherID) {
...
                 let output = {
-                    "refundedVoucherID":burntVoucherID
+                    "refundedVoucherID":burntVoucherID,
+                    "gasPaid":gasUsed,
+                    "gasUsed":gasUsed
                 }
 
-                console.log(output)
-                return(output)
+                // console.log(output)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/refundVoucher.js]: removed

File: testGameTree/helpers/checkBalance.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+        table.push(["SELLER",SELLER_PUBLIC,sbal]);
+        table.push(["BUYER",BUYER_PUBLIC,bbal]);
+
+        return(table.toString());
+}
+
+module.exports = checkBalances;
+
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/helpers/checkBalance.js:27]: resolved

File: testGameTree/helpers/formatter.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// Formats the input data into a tabular format
+let Table = require('cli-table');
+
+async function formatter(_input) {
+    // console.log(_input);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/helpers/formatter.js:5]: removed the commented line

File: testGameTree/helpers/formatter.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    // table is an Array, so you can `push`, `unshift`, `splice` and friends
+    for(let i=0;i<listArray.length;i++){
+        table.push(listArray[i]);
+    }
+    console.log(table.toString());
+}
+
+module.exports = formatter;
+
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/helpers/formatter.js:19]: removed the commented lines

File: testGameTree/seller/cancelVoucherSet.js [hunk 8]
```diff
@@ -17,11 +17,11 @@ Contract.setProvider(PROVIDER);
 // const seller = new ethers.Wallet(SELLER_SECRET, PROVIDER);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/cancelVoucherSet.js]: removed

File: testGameTree/seller/cancelVoucherSet.js [hunk 9]
```diff
@@ -17,11 +17,11 @@ Contract.setProvider(PROVIDER);
...
 const seller = SELLER_PUBLIC;
 
-// let sellerRequestCancelorFault =
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/cancelVoucherSet.js:20]: removed

File: testGameTree/seller/cancelVoucherSet.js [hunk 10]
```diff
@@ -49,36 +49,42 @@ const seller = SELLER_PUBLIC;
                     reject(new Error(err.message))
                 }
 
-                resolve(hash)
-            }).on('transactionHash', function(hash){
-                console.log("Transaction Hash : "+hash);
+                console.log("Transaction Hash : ",hash);
             }).on('receipt', function(receipt){
                 // console.log(receipt)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/cancelVoucherSet.js]: removed

File: testGameTree/seller/cancelVoucherSet.js [hunk 11]
```diff
@@ -49,36 +49,42 @@ const seller = SELLER_PUBLIC;
...
+                    "gasUsed":gasUsed,
+                    "redfundedSellerDeposit":redfundSellerDeposit,
+                    "logReceipt1": receipt.logs[0].id,
+                    "logReceipt2": receipt.logs[1].id,
+                    "logReceipt3": receipt.logs[2].id
                 }
 
-                console.log(output)
-                return(output)
+                // console.log(output)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/cancelVoucherSet.js]: removed

File: testGameTree/seller/cancelVoucherSet.js [hunk 12]
```diff
@@ -49,36 +49,42 @@ const seller = SELLER_PUBLIC;
...
 }
 
-(async function newOrder () {
-    await requestCancelorFault("57896044618658097711785492504343954004219371990794251689378202498399717031936");
-})();
+// (async function newOrder () {
+//     await requestCancelorFault("57896044618658097711785492504343954004219371990794251689378202498399717031936");
+// })();
 
-// module.exports = sellerRequestCancelorFault;
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/cancelVoucherSet.js]: removed

File: testGameTree/seller/createVoucher.js [hunk 13]
```diff
@@ -0,0 +1,103 @@
+let Web3 = require('web3');
+const BN = require('bn.js');
+let Contract = require('web3-eth-contract');
+const helpers = require('../helpers/constants')
+const Tx = require('ethereumjs-tx').Transaction;
+let converter = require('hex2dec');
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/createVoucher.js]: removed

File: testGameTree/seller/createVoucher.js [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+const BosonRouter = require("../../build/contracts/BosonRouter.json").abi;
+const { SELLER_SECRET, SELLER_PUBLIC, contracts, PROVIDER } = require('../helpers/config');
+
+let web3 = new Web3(new Web3.providers.HttpProvider(PROVIDER));
+
+// set provider for all later instances to use
+Contract.setProvider(PROVIDER);
+
+// const seller = new ethers.Wallet(SELLER_SECRET, PROVIDER);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/createVoucher.js]: removed

File: testGameTree/seller/createVoucher.js [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+
+            // executes the transaction
+            web3.eth.sendSignedTransaction('0x' + serializedTx.toString('hex'), (err, hash) => {
+                if(err) {
+                    console.log(err)
+                    reject(new Error(err.message))
+                }
+
+                console.log("Transaction Hash : ",hash);
+                // resolve(hash)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/createVoucher.js]: removed

File: testGameTree/seller/createVoucher.js [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+            }).on('receipt', function(receipt){
+                // console.log(receipt);
+                // console.log(receipt.topics[1]);
+                // console.log(receipt.topics[2]);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/createVoucher.js]: removed

File: testGameTree/seller/createVoucher.js [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+                    "nftSeller":"0x"+nftSeller,
+                    "gasPaid": converter.hexToDec(gasSent),
+                    "gasUsed":gasUsed,
+                    "sellerDeposit":helpers.PROMISE_PRICE1,
+                    "logReceipt1": receipt.logs[0].id,
+                    "logReceipt2": receipt.logs[1].id,
+                    "logReceipt3": receipt.logs[2].id
+                }
+
+                // console.log(output)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/createVoucher.js]: removed

File: testGameTree/seller/createVoucher.js [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+                resolve(output)
+            }).on('error', console.error);
+        })
+    })
+}
+
+// (async function newOrder () {
+//     await CreateOrderETHETH();
+// })();
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/createVoucher.js]: removed

File: testGameTree/seller/faultVoucher.js [hunk 19]
```diff
@@ -45,33 +44,37 @@ function faultVoucher(_voucherSetID) {
             // executes the transaction
             web3.eth.sendSignedTransaction('0x' + serializedTx.toString('hex'), (err, hash) => {
                 if(err) {
-                    console.log(err)
+                    // console.log(err)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/faultVoucher.js]: removed

File: testGameTree/seller/faultVoucher.js [hunk 20]
```diff
@@ -45,33 +44,37 @@ function faultVoucher(_voucherSetID) {
...
                     reject(new Error(err.message))
                 }
 
-                resolve(hash)
-            }).on('transactionHash', function(hash){
+                // resolve(hash)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/faultVoucher.js]: removed

File: testGameTree/seller/faultVoucher.js [hunk 21]
```diff
@@ -45,33 +44,37 @@ function faultVoucher(_voucherSetID) {
...
+                    "TransactionHash":txHash,
                     "FaultedVoucherID":FaultedVoucherID,
+                    "gasPaid":converter.hexToDec(gasPaid),
+                    "gasUsed":gasUsed,
+                    "status":txStatus
                 }
 
-                console.log(output)
-                return(output)
+                // console.log(output)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/faultVoucher.js]: removed

File: testGameTree/seller/faultVoucher.js [hunk 22]
```diff
@@ -45,33 +44,37 @@ function faultVoucher(_voucherSetID) {
...
 
-(async function newOrder () {
-    await faultVoucher("57896044618658097711785492504343954004219371990794251689378202498399717031936");
-})();
-
-// module.exports = sellerfalutVoucher;
-
+// (async function newOrder () {
+//     await faultVoucher("57896044618658097711785492504343954004219371990794251689378202498399717031936");
+// })();
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/faultVoucher.js]: removed

File: testGameTree/seller/faultVoucher.js [hunk 23]
```diff
@@ -45,33 +44,37 @@ function faultVoucher(_voucherSetID) {
...
+                    // console.log(err)
                     reject(new Error(err.message))
                 }
 
-                resolve(hash)
-            }).on('transactionHash', function(hash){
+                // resolve(hash)
                 console.log("Transaction Hash : "+hash);
             }).on('receipt', function(receipt){
                 // console.log(receipt)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/seller/faultVoucher.js]: removed

File: testGameTree/2-seller-create-cancel.js [hunk 24]
```diff
@@ -0,0 +1,34 @@
+const sellerCreate = require("./seller/createVoucher");
+const sellerCancel = require("./seller/cancelVoucherSet");
+const {describe,it} = require("mocha");
+
+let format = require("./helpers/formatter");
+const checkBalance = require("./helpers/checkBalance");
+
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/2-seller-create-cancel.js:8]: removed

File: testGameTree/4-buyer-commit-redeem.js [hunk 25]
```diff
@@ -0,0 +1,48 @@
+const sellerCreate = require("./seller/createVoucher");
+const commitVocucher = require("./buyer/commitVoucher");
+const redeemVoucher = require("./buyer/redeemVoucher");
+const checkBalance = require("./helpers/checkBalance");
+const delay = require("./helpers/delay");
+const {describe,it} = require("mocha");
+
+let format = require("./helpers/formatter")
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/4-buyer-commit-redeem.js]: removed

File: testGameTree/6-buyer-commit-redeem-complain.js [hunk 26]
```diff
@@ -0,0 +1,56 @@
+const sellerCreate = require("./seller/createVoucher");
+const commitVocucher = require("./buyer/commitVoucher");
+const redeemVoucher = require("./buyer/redeemVoucher");
+const checkBalance = require("./helpers/checkBalance");
+const complainVoucher = require("./buyer/compainVoucher");
+const delay = require("./helpers/delay");
+const {describe,it} = require("mocha");
+
+let format = require("./helpers/formatter");
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/6-buyer-commit-redeem-complain.js]: removed

File: testGameTree/7-buyer-commit-refund-complain.js [hunk 27]
```diff
@@ -0,0 +1,55 @@
...
+        committedVoucher = await commitVocucher(voucherSetDetails["createdVoucherSetID"]);
+        await format(committedVoucher);
+    });
+
+    it("TEST SCENARIO 07 :: BUYER REFUNDS :: 3. Buyer refunds a purchased voucher", async function() {
+        await delay();
+        console.log(await checkBalance());
+        let refund = await refundVoucher(committedVoucher["MintedVoucherID"]);
+        await format(refund);
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/7-buyer-commit-refund-complain.js]: removed

File: testGameTree/8-buyer-commit-redeem-complain-seller-fault.js [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+const complainVoucher = require("./buyer/compainVoucher");
+const faultVoucher = require("./seller/faultVoucher");
+const delay = require("./helpers/delay");
+const {describe,it} = require("mocha");
+let format = require("./helpers/formatter");
+
+const TIMEOUT = 500 * 1000;
+
+describe("TEST SCENARIO 08 :: SELLER CREATES, BUYER COMMITS, REDEEMS & COMPLAINS, SELLER FAULTS", async function() {
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/8-buyer-commit-redeem-complain-seller-fault.js:17]: removed

File: testGameTree/9-buyer-commit-refund-complain-seller-fault.js [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+const commitVocucher = require("./buyer/commitVoucher");
+const refundVoucher = require("./buyer/refundVoucher");
+const checkBalance = require("./helpers/checkBalance");
+const complainVoucher = require("./buyer/compainVoucher");
+const faultVoucher = require("./seller/faultVoucher");
+const delay = require("./helpers/delay");
+const {describe,it} = require("mocha");
+let format = require("./helpers/formatter");
+
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/9-buyer-commit-refund-complain-seller-fault.js:13]: removed

File: testGameTree/9-buyer-commit-seller-fault.js [hunk 30]
```diff
@@ -0,0 +1,42 @@
+const sellerCreate = require("./seller/createVoucher");
+// const sellerCancel = require("./seller/cancelVoucherSet");
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/9-buyer-commit-seller-fault.js]: removed

File: testGameTree/9-buyer-commit-seller-fault.js [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+const commitVocucher = require("./buyer/commitVoucher");
+const faultVoucher = require("./seller/faultVoucher");
+const delay = require("./helpers/delay");
+const checkBalance = require("./helpers/checkBalance");
+const {describe,it} = require("mocha");
+
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/9-buyer-commit-seller-fault.js]: removed

File: testGameTree/9-buyer-commit-seller-fault.js [hunk 32]
```diff
@@ -0,0 +1,42 @@
...
+
+const TIMEOUT = 500 * 1000;
+
+describe("TEST SCENARIO 01 :: SELLER CREATE", async function() {
+
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/9-buyer-commit-seller-fault.js]: removed

File: testGameTree/buyer/commitVoucher.js [hunk 33]
```diff
@@ -50,13 +49,14 @@ const buyer = BUYER_PUBLIC;
                     console.log(err)
                     reject(new Error(err.message))
                 }
-                resolve(hash)
-            }).on('transactionHash', function(hash){
+                // resolve(hash)
                 console.log("Transaction Hash : "+hash);
             }).on('receipt', function(receipt){
                 // console.log(receipt);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/commitVoucher.js:45]: removed

File: testGameTree/buyer/commitVoucher.js [hunk 34]
```diff
@@ -65,29 +65,32 @@ const buyer = BUYER_PUBLIC;
...
                 let output = {
+                    "TransactionHash":txhash,
                     "VoucherSetID":voucherSetID,
                     "MintedVoucherID":mintedVoucherID,
                     "issuer":"0x"+issuer,
                     "holder":"0x"+holder,
                     "promiseID":promiseID,
+                    "gasPaid":converter.hexToDec(gasPaid),
+                    "gasUsed":gasUsed,
                     "correlationID":correlationID,
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/commitVoucher.js:65]: mofied

File: testGameTree/buyer/compainVoucher.js [hunk 35]
```diff
@@ -16,20 +16,19 @@ Contract.setProvider(PROVIDER);
 
 const buyer = BUYER_PUBLIC;
 
-// let ComplainVoucherETHETH =
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/compainVoucher.js:19]: removed

File: testGameTree/buyer/compainVoucher.js [hunk 36]
```diff
@@ -47,29 +46,30 @@ function complainVoucher(_voucherID) {
                     console.log(err)
                     reject(new Error(err.message))
                 }
-                resolve(hash)
-            }).on('transactionHash', function(hash){
                 console.log("Transaction Hash : "+hash);
             }).on('receipt', function(receipt){
                 // console.log(receipt);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/compainVoucher.js]: removed

File: testGameTree/buyer/compainVoucher.js [hunk 37]
```diff
@@ -47,29 +46,30 @@ function complainVoucher(_voucherID) {
...
                 let output = {
-                    "complainedVoucherID":complainedVoucherID
+                    "complainedVoucherID":complainedVoucherID,
+                    "gasPaid":converter.hexToDec(gasPaid),
+                    "gasUsed":gasUsed
                 }
 
-                console.log(output)
-                return(output)
+                // console.log(output)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/compainVoucher.js]: removed

File: testGameTree/buyer/compainVoucher.js [hunk 38]
```diff
@@ -47,29 +46,30 @@ function complainVoucher(_voucherID) {
...
 }
 
-(async function newOrder () {
-    await complainVoucher("57896044618658097711785492504343954004559654357715190152841577105831485243399");
-})();
+// (async function newOrder () {
+//     await complainVoucher("57896044618658097711785492504343954004559654357715190152841577105831485243399");
+// })();
 
-// module.exports = ComplainVoucherETHETH;
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/compainVoucher.js]: removed

File: testGameTree/buyer/redeemVoucher.js [hunk 39]
```diff
@@ -1,13 +1,13 @@
 let Web3 = require('web3');
-const BN = require('bn.js');
+// const BN = require('bn.js');
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/redeemVoucher.js]: removed

File: testGameTree/buyer/redeemVoucher.js [hunk 40]
```diff
@@ -1,13 +1,13 @@
...
 let Contract = require('web3-eth-contract');
-const helpers = require('../helpers/constants')
+// const helpers = require('../helpers/constants')
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/redeemVoucher.js]: removed

File: testGameTree/buyer/redeemVoucher.js [hunk 41]
```diff
@@ -1,13 +1,13 @@
...
 const Tx = require('ethereumjs-tx').Transaction;
 let converter = require('hex2dec');
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/redeemVoucher.js]: removed

File: testGameTree/buyer/redeemVoucher.js [hunk 42]
```diff
@@ -16,20 +16,19 @@ Contract.setProvider(PROVIDER);
 
 const buyer = BUYER_PUBLIC;
 
-// let RedeemVoucherETHETH =
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/redeemVoucher.js:19]: removed

File: testGameTree/buyer/redeemVoucher.js [hunk 43]
```diff
@@ -47,34 +46,36 @@ const buyer = BUYER_PUBLIC;
                     console.log(err)
                     reject(new Error(err.message))
                 }
-                resolve(hash)
-            }).on('transactionHash', function(hash){
+                // resolve(hash)
                 console.log("Transaction Hash : "+hash);
             }).on('receipt', function(receipt){
                 // console.log(receipt);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/redeemVoucher.js]: removed

File: testGameTree/buyer/redeemVoucher.js [hunk 44]
```diff
@@ -47,34 +46,36 @@ const buyer = BUYER_PUBLIC;
...
                     "holder":"0x"+holder,
-                    "promiseID":promiseID
+                    "promiseID":promiseID,
+                    "gasPaid":gasPaid,
+                    "gasUsed":gasUsed
                 }
 
-                console.log(output)
-                return(output)
+                // console.log(output)
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/redeemVoucher.js]: removed

File: testGameTree/buyer/redeemVoucher.js [hunk 45]
```diff
@@ -47,34 +46,36 @@ const buyer = BUYER_PUBLIC;
...
 }
 
-(async function newOrder () {
-    await redeemVoucher("57896044618658097711785492504343954004559654357715190152841577105831485243399");
-})();
+// (async function newOrder () {
+//     await redeemVoucher("57896044618658097711785492504343954004559654357715190152841577105831485243399");
+// })();
 
-// module.exports = RedeemVoucherETHETH;
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/redeemVoucher.js]: removed

File: testGameTree/buyer/refundVoucher.js [hunk 46]
```diff
@@ -47,29 +47,32 @@ function refundVoucher(_voucherID) {
                     console.log(err)
                     reject(new Error(err.message))
                 }
-                resolve(hash)
-            }).on('transactionHash', function(hash){
+                // resolve(hash)
+                // console.log(hash);
```
[the remark this answers is not in the document]
Comment (sriharikapu) [reply testGameTree/buyer/refundVoucher.js]: removed

**Comment by sriharikapu** [comment]
Why the old test were deleted? There were roughly about 400 tests. Do these scenarios cover each and every missing test?
 - [x] Brought Back the previous test
 
 Agreed with Heather that the actions folder needs to be moved into the tests, maybe renamed to testGameTreeActions or similar
 - [x] Renamed to testGameTree
 
 Agreed with Heather that the tests should be using assertions
 - [x] Added More assertions to tests
 
Do we need to test those actions over a testnet at all?
 - [x] You can set the provider to local or any testnet url so you can run them
 
There should be something at the end indicating that the tests are completed. Currently when the process finishes I was misled that I am doing something wrong as there is not anything which prompts me how tests went
- [x] once the tests finish you will be able to see the number of tests passing, 
 
Maybe a little more detailed README file would be good. I don't think the delay should be that long (30000) if we test the flow locally. Maybe even we could move it into the configs?
- [x] delay was set to 30000ms initially because of the delay in transactions getting confirmed lately, I had kept the delay to the config so you can customized for local test and testnet etc

constant.js exported from the test folder, and now moved to actions/helpers/constants. Do we need to do that. IMHO this should be in the test folder as it's own purpose is to serve the tests
- [x] Actually we don't need all the variables in the constants from the `constants.js`, I had removed the unused variables. also we can link to the previous constants

left some comments for removing the empty lines, commented lines and other
- [x] removed unnecessary empty lines mentioned

pipeline should succeed before merging
- [x]  made sure that pipeline passes

**Review by hswopeams — APPROVED** [review]

**Comment by sriharikapu** [comment]
@hswopeams I think we can merge this once @HristiyanG approves it. Also I checked at my end everything works as expected.

**Comment by hswopeams** [comment]
Changed gametree tests to run locally on Ganace instance started by the (Ruby) `go` script. Discussed with Holly, and it was decided that running locally is enough. There's no need to add complexity by adding config to also run against a testnet.

**Comment by thecryptofruit** [comment]
@damir-bajramovic , please have a look if this is in your card, otherwise close. thanks

**Comment by mischat** [comment]
All of the Game Tree has been incorporated into our unit test suite, we are happy that we are covering all of the game tree and more in `main` of this repo. This branch will be deleted.

**Pull request closed without merging**